### PR TITLE
add text in the pruning images section for --skip-registry option

### PR DESCRIPTION
**Pull request by gabemontero**
Doc for https://trello.com/c/V1anqimn/1418-5-image-soft-prune-imagestreams

@openshift/sig-developer-experience @bmcelvee ptal

disclaimer:  I sided on full transparency for this initial stab, trying to capture the warnings/concerns/recommendations that surfaced during the card planning and pr review activities.  Certainly during the 
review here, aside form massaging the wording to get it as good as possible, if the details are not appropriate for this context (perhaps better suited to the yet to be crafted, more comprehensive general image pruning recommendation/best-practices guide that @bparees cited), we can of course "prune" ;-) as needed.

### CODE REVIEW

**Comment by gabemontero** [comment]
[(5) Image soft prune [imagestreams]](https://trello.com/c/V1anqimn/1418-5-image-soft-prune-imagestreams)

**Comment by bmcelvee** [comment]
@gabemontero, thanks! For the most part, this looks good to me. I would suggest rewording the note a bit, something along the lines of:

"Separating the removal of {product-title} Image API Objects and Image data
from the Registry by using `--skip-registry` followed by xref:#hard-pruning-registry[Hard Pruning the Registry] narrows some timing windows, and is safer when compared to trying
to prune both through one command. However, timing windows are not completely removed. 

For example, you can still create a Pod referencing an Image as pruning identifies that Image for pruning. You should still keep track of an API Object created during the pruning operations that might reference Images, so you can mitigate any references to deleted content."

Does it still convey the warning properly?

**Comment by gabemontero** [comment]
pushed update for both flip of name/default for arg, as well as @bmcelvee 's suggestion.

File: admin_guide/pruning_resources.adoc [hunk 1]
```diff
@@ -235,6 +235,16 @@ case it fails (the registry cannot be resolved or reached), an alternative
 route that works needs to be provided using this flag. The registry host name
 may be prefixed by `https://` or `http://` which will enforce particular
 connection protocol.
+
+.^|`--prune-registry`
+|In conjunction with the conditions stipulated by the other options, this option controls
+the operation only considers the {product-title} Image API Objects, or also processes
```
Comment (bparees) [inline admin_guide/pruning_resources.adoc]: `this option controls the operation only considers the `

should be "this option controls whether the"?

File: admin_guide/pruning_resources.adoc [hunk 2]
```diff
@@ -235,6 +235,16 @@ case it fails (the registry cannot be resolved or reached), an alternative
...
+the corresponding data in the registry.  By default, image pruning processes both the
+Image API Objects and corresponding data in the registry.
+This option makes most sense when there is either a large amount of data in etcd
+and/or a large number of images or blobs. After running with this option set to `false`, you can
```
Comment (bparees) [inline admin_guide/pruning_resources.adoc]: I would phrase it as "This options is useful when you are only concerned with removing etcd content, possibly to reduce the number of image objects, but are not concerned with cleaning up registry storage, or intend to do that separately via hard prune, possibly during an appropriate maintenance window for the registry."

File: admin_guide/pruning_resources.adoc [hunk 3]
```diff
@@ -278,6 +288,20 @@ image streams that have a reference to the image in `*status.tags*`.
...
+Separating the removal of {product-title} Image API Objects and Image data
+from the Registry by using `--prune-registry=false` followed by
+xref:#hard-pruning-registry[Hard Pruning the Registry] narrows some timing windows,
+and is safer when compared to trying to prune both through one command. However,
+timing windows are not completely removed.
+
+For example, you can still create a Pod referencing an Image as pruning identifies that
+Image for pruning. You should still keep track of an API Object created during the pruning
+operations that might reference Images, so you can mitigate any references to deleted
+content.
```
Comment (bparees) [inline admin_guide/pruning_resources.adoc]: +1, but @legionus may have some additional comments on what we should say here.

Comment (legionus) [reply admin_guide/pruning_resources.adoc]: Please add a warning that re-doing the pruning without `--prune-registry` option or with `--prune-registry=true` will not lead to pruning the storage of image registry. I mean that all images that was pruned with `--prune-registry=false` can be deleted from the registry storage only by using the hard-pruning otherwise the space will be lost.

Comment (bparees) [reply admin_guide/pruning_resources.adoc]: Good catch!

**Comment by gabemontero** [comment]
update pushed for @bparees and @legionus suggestions

**Comment by bparees** [comment]
lgtm

File: admin_guide/pruning_resources.adoc [hunk 4]
```diff
@@ -235,6 +235,16 @@ case it fails (the registry cannot be resolved or reached), an alternative
 route that works needs to be provided using this flag. The registry host name
 may be prefixed by `https://` or `http://` which will enforce particular
 connection protocol.
+
+.^|`--prune-registry`
+|In conjunction with the conditions stipulated by the other options, this option controls
+whethers the data in the registry corresponding to the {product-title} Image API Objects
```
Comment (bmcelvee) [inline admin_guide/pruning_resources.adoc]: s/whethers/whether

File: admin_guide/pruning_resources.adoc [hunk 5]
```diff
@@ -235,6 +235,16 @@ case it fails (the registry cannot be resolved or reached), an alternative
...
+is pruned.  By default, image pruning processes both the
```
Comment (bmcelvee) [inline admin_guide/pruning_resources.adoc]: Double space after "...pruned."

File: admin_guide/pruning_resources.adoc [hunk 6]
```diff
@@ -235,6 +235,16 @@ case it fails (the registry cannot be resolved or reached), an alternative
...
+Image API Objects and corresponding data in the registry.
+This options is useful when you are only concerned with removing etcd content, possibly
+to reduce the number of image objects, but are not concerned with cleaning up registry
+storage, or intend to do that separately via xref:#hard-pruning-registry[Hard Pruning the Registry],
```
Comment (bmcelvee) [inline admin_guide/pruning_resources.adoc]: s/registry storage,/registry storage;  --Does that make more sense?

Comment (bmcelvee) [inline admin_guide/pruning_resources.adoc]: s/via/by

File: admin_guide/pruning_resources.adoc [hunk 7]
```diff
@@ -278,6 +288,26 @@ image streams that have a reference to the image in `*status.tags*`.
...
+timing windows are not completely removed.
+
+For example, you can still create a Pod referencing an Image as pruning identifies that
+Image for pruning. You should still keep track of an API Object created during the pruning
+operations that might reference Images, so you can mitigate any references to deleted
+content.
+
+Also, keep in mind that re-doing the pruning without the `--prune-registry` option or with
+`--prune-registry=true` will not lead to pruning the associated storage in the image registry
+for image previously pruned via `--prune-registry=false`.
```
Comment (bmcelvee) [inline admin_guide/pruning_resources.adoc]: s/via/by

Comment (bmcelvee) [inline admin_guide/pruning_resources.adoc]: s/image previously/images previously

Comment (bmcelvee) [reply admin_guide/pruning_resources.adoc]: or "an image"

File: admin_guide/pruning_resources.adoc [hunk 8]
```diff
@@ -278,6 +288,26 @@ image streams that have a reference to the image in `*status.tags*`.
...
+Any images that were pruned with `--prune-registry=false` can only be deleted from
+registry storage via xref:#hard-pruning-registry[Hard Pruning the Registry].
```
Comment (bmcelvee) [inline admin_guide/pruning_resources.adoc]: s/via/by

**Comment by bmcelvee** [comment]
Just a few suggestions from me. Otherwise, looks good!

**Comment by gabemontero** [comment]
thanks @bmcelvee updates pushed 

note, let's not merge into latest until https://github.com/openshift/origin/pull/17480 merges

**Comment by bmcelvee** [comment]
Thanks, @gabemontero! Sounds good :)

**Comment by gabemontero** [comment]
hey @bmcelvee https://github.com/openshift/origin/pull/17480 has merged

**Comment by bmcelvee** [comment]
Thanks, @gabemontero! This is going into 3.9 correct?

**Comment by bmcelvee** [comment]
[rev_history]
|xref:..admin_guide/pruning_resources.adoc#admin-guide-pruning-resources[Pruning Objects]
|Added `--prune-registry` CLI configuration option and corresponding note to the xref:..admin_guide/pruning_resources.adoc#pruning-images[Pruning Images] section.
%

**Comment by gabemontero** [comment]
On Wed, Dec 6, 2017 at 10:46 AM, Brandi McElveen Munilla <
notifications@github.com> wrote:

> Thanks, @gabemontero <https://github.com/gabemontero>! This is going into
> 3.9 correct?
>

yep 3.9
